### PR TITLE
Link extension

### PR DESCRIPTION
**Pull request by mharsat**
Link extension features:
- Automatically create links from text that represents a link (on enter/space)
- Create links from pasted text
- Add link decoration to text when pasting to it

Also:
- Added prosemirror dev utils to tiptap editor, works in storybook
- Nodes split from other nodes have `id: ''` instead of duplicating `id`
- Recovered the `RicosTiptapEditor.spec` test

### CODE REVIEW

File: packages/plugin-link-preview/web/lib/utils.ts [hunk 1]
```diff
@@ -13,48 +13,90 @@ import {
...
-    const { newEditorState } = createBlock(withoutLinkBlock, data, LINK_PREVIEW_TYPE);
-    setEditorState(newEditorState);
+    const linkPreviewData = await createLinkPreviewData(
+      linkData,
+      linkPreview,
+      config[LINK_PREVIEW_TYPE]
+    );
+    addLinkPreviewBlock(editorState, config, blockKey, linkPreviewData);
+  }
+};
```
Comment (mharsat) [inline packages/plugin-link-preview/web/lib/utils.ts:44]: @shaulgo @danarish4 FYI I split `addLinkPreview` to make it more modular as a preparation for using it in tiptap without draft code

File: package.json [hunk 2]
```diff
@@ -175,6 +175,7 @@
     "postcss-url": "^7.3.2",
     "prompts": "^2.0.1",
     "prop-types": "^15.6.2",
+    "prosemirror-dev-tools": "^3.1.0",
```
Comment (Barackos) [inline package.json:178]: Awesome 🎉

File: packages/plugin-link-preview/web/src/tiptap/tiptap.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export const tiptapExtensions = [
+  createNodeExtension({
+    Component,
+    createComponentDataDefaults: ({ LinkPreviewData }) => LinkPreviewData.fromJSON({}),
+    createConfig: () => ({
+      name,
+      atom: false,
+      addCommands() {
+        return {};
+      },
```
Comment (Barackos) [inline packages/plugin-link-preview/web/src/tiptap/tiptap.ts:21]: `addCommands` is optional
```suggestion
```

**Review by Barackos — APPROVED** [review]

File: packages/plugin-link-preview/web/src/tiptap/tiptap.ts (hunk 3 above)
Comment (mharsat) [reply packages/plugin-link-preview/web/src/tiptap/tiptap.ts:21]: It's still WIP, will probably need it

**Pull request merged**